src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -207,7 +213,28 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
 
             return metadata;
         }
+        
+        /// <inheritdoc />
+        public override async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool useCache = true)
+        {
+            Logger.Trace("PackageExists {0}", purl?.ToString());
+            if (purl is null)
+            {
+                Logger.Trace("Provided PackageURL was null.");
+                return false;
+            }
+
+            if(purl.Version.IsBlank())
+            {
+                Logger.Trace("Provided PackageURL version was null or blank.");
+                return false;
+            }
 
+            // NuGet packages are case insensitive. Convert purl to lowercase before comparing against package versions list. 
+            var versionLowercase = purl.Version.ToLowerInvariant();
+            return (await EnumerateVersionsAsync(purl, useCache)).Contains(versionLowercase);
```

Comment:
Rather than converting the string to lowercase since the lowered string is only used in this call I think it would be better to use the Linq Contains with the appropriate string comparer. 

https://learn.microsoft.com/en-us/dotnet/api/system.stringcomparer.invariantcultureignorecase?view=net-7.0